### PR TITLE
[FAN-46] Transfer and burn token

### PR DESCRIPTION
**Pull request by Denise54321**

### CODE REVIEW

File: src/modules/brand/components/BurnTokenDialog/BurnTokenDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+  onClose,
+  onSubmit,
+}: {
+  loading: boolean;
+  isOpen: boolean;
+  standard: NftType;
+  maxQuantity?: number;
+  onClose: () => void;
+  onSubmit: (value: IBurnFormValues, form: any, callback: any) => void;
+}) => {
```
Comment (Antroll) [inline src/modules/brand/components/BurnTokenDialog/BurnTokenDialog.tsx]: Better to move the typing into a named interface `IBurnTokenDialogProps` like other copmonents in the project.

File: src/modules/brand/components/TransferTokenDialog/TransferTokenDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+  onClose,
+  onSubmit,
+}: {
+  loading: boolean;
+  isOpen: boolean;
+  standard: NftType;
+  maxQuantity?: number;
+  onClose: () => void;
+  onSubmit: (value: ITransferFormValues, form: any, callback: any) => void;
+}) => {
```
Comment (Antroll) [inline src/modules/brand/components/TransferTokenDialog/TransferTokenDialog.tsx]: Better to move the typing into a named interface `ITransferTokenDialogProps` like other copmonents in the project.

File: src/modules/common/actions/burnToken.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+import { Store } from 'redux';
+import { setAccount } from 'modules/account/store/actions/setAccount';
+import {
+  BounceErc721,
+  BounceErc1155,
+} from '../../web3/contracts';
+import { NftType } from 'modules/createNFT/actions/createNft';
+import { TransactionReceipt } from '@ethersproject/abstract-provider';
+
+export const burnToken = createSmartAction(
```
Comment (Antroll) [inline src/modules/common/actions/burnToken.ts]:
```suggestion
export const burnToken = createSmartAction<RequestAction<Data, TransformedData>>(
```
I think need to write typings for that.

File: src/modules/common/actions/transferToken.ts [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+import { Store } from 'redux';
+import { setAccount } from 'modules/account/store/actions/setAccount';
+import {
+  BounceErc721,
+  BounceErc1155,
+} from '../../web3/contracts';
+import { NftType } from 'modules/createNFT/actions/createNft';
+import { TransactionReceipt } from '@ethersproject/abstract-provider';
+
+export const transferToken = createSmartAction(
```
Comment (Antroll) [inline src/modules/common/actions/transferToken.ts]: The same as in the previous action.
`RequestAction<Data, TransformedData>`

File: src/modules/common/actions/burnToken.ts (hunk 3 above)
Comment (Denise54321) [reply src/modules/common/actions/burnToken.ts]: This action is working with the smart contract, it didn't have any type. Maybe we can put <RequestAction<void, void>>, I think we also can ignore it.

File: src/modules/common/actions/transferToken.ts (hunk 4 above)
Comment (Denise54321) [reply src/modules/common/actions/transferToken.ts]: The same as before.

File: src/modules/brand/components/BurnTokenDialog/BurnTokenDialog.tsx (hunk 1 above)
Comment (BigBadAlien) [reply src/modules/brand/components/BurnTokenDialog/BurnTokenDialog.tsx]:
```suggestion
interface IBurnTokenDialogProps {
  loading: boolean;
  isOpen: boolean;
  standard: NftType;
  maxQuantity?: number;
  onClose: () => void;
  onSubmit: (value: IBurnFormValues, form: any, callback: any) => void;
}

export const BurnTokenDialog = ({
  loading,
  isOpen,
  standard,
  maxQuantity,
  onClose,
  onSubmit,
}: IBurnTokenDialogProps) => {
```

File: src/modules/common/actions/burnToken.ts (hunk 3 above)
Comment (BigBadAlien) [reply src/modules/common/actions/burnToken.ts]:
```
<RequestAction<void, void>>
```

void, void is ok

**Review by Antroll — APPROVED** [review]

**Review by BigBadAlien — APPROVED** [review]

**Pull request merged**